dot/state/epoch.go

```diff
@@ -476,10 +476,35 @@ func (s *EpochState) GetStartSlotForEpoch(epoch uint64, bestBlockHash common.Has
 	return s.epochLength*epoch + chainFirstSlotNumber, nil
 }
 
+// retrieveFirstNonOriginBlockSlot returns the slot number of the first non origin block
+func (s *EpochState) retrieveFirstNonOriginBlockSlotFromDb(hash common.Hash) (uint64, error) {
```

Comment:
Instead of having the db get here, I think that we can have a method at `BlockState` like `SetFirstNonOriginSlotNumber` but for retrieve eg. `GetFirstNonOriginSlotNumber`, then we call `s.blockState.GetFirstNonOriginSlotNumber()`

Reply:
Nice!
I was actually wondering if the epochState should have that receiver  thanks!.